apex-checks/src/main/java/org/fundacionjala/enforce/sonarqube/apex/checks/DeprecatedMethodCheck.java

```diff
@@ -0,0 +1,125 @@
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+import org.sonar.squidbridge.checks.SquidCheck;
+
+import org.fundacionjala.enforce.sonarqube.apex.api.ApexKeyword;
+import org.fundacionjala.enforce.sonarqube.apex.api.grammar.RuleKey;
+
+/**
+ * This class checks that a deprecated method does not contain lines of code.
+ */
+@Rule(
+        key = DeprecatedMethodCheck.CHECK_KEY,
+        priority = Priority.INFO,
+        name = "Verification methods deprecated",
+        description = "Prevent the body of a method contains deprecated code lines",
+        tags = Tags.OBSOLETE
+)
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
+@SqaleConstantRemediation("1min")
+@ActivatedByDefault
+public class DeprecatedMethodCheck extends SquidCheck<Grammar> {
+
+    /**
+     * Stores a message template.
+     */
+    public static final String MESSAGE = "The \"%s\" method is deprecated, suggest deleting the contents of the method";
+
+    /**
+     * It is the code of the rule for the plugin.
+     */
+    public static final String CHECK_KEY = "A1006";
+
+    /**
+     * The variables are initialized and subscribe the base rule.
+     */
+    @Override
+    public void init() {
+        subscribeTo(RuleKey.METHOD_DECLARATION);
+    }
+
+    /**
+     * It is responsible for verifying whether the rule is met in the rule base.
+     * In the event that the rule is not correct, create message error.
+     *
+     * @param astNode It is the node that stores all the rules.
+     */
+    @Override
+    public void visitNode(AstNode astNode) {
+        if (astNode.hasDescendant(RuleKey.ANNOTATION)) {
+            List<AstNode> childrensMethodDeclaration = astNode.getChildren();
+            if (validateToken(childrensMethodDeclaration.get(0).getChildren())) {
+                AstNode nameMethod = validateStatementBlock(childrensMethodDeclaration);
+                if (nameMethod != null) {
+                    getContext().createLineViolation(this, String.format(MESSAGE,
+                            nameMethod.getTokenValue()), astNode);
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks if node token type is deprecated.
+     *
+     * @param astNode is the node to analyze.
+     * @return A value of true, if the node is expected.
+     */
+    private boolean validateToken(List<AstNode> astNode) {
+        return astNode.get(1).getTokenValue().equals(ApexKeyword.DEPRECATED.getValue());
+    }
+
+    /**
+     * Checks if the node BLOCK_STATEMENT has content.
+     *
+     * @param astNode is the node to analyze.
+     * @return If the node has content, returns the same node, null if has not.
+     */
+    private AstNode validateStatementBlock(List<AstNode> astNode) {
```

Comment:
too much work to verify if the method has body inside, please review and improve the validation